static/app/views/automations/components/automationBuilderContext.tsx

```diff
@@ -395,6 +446,96 @@ function updateIfCondition(
   };
 }
 
+function addIfAction(
+  state: AutomationBuilderState,
+  action: AddIfActionAction
+): AutomationBuilderState {
+  const {groupIndex, actionType} = action;
+  return {
+    ...state,
+    actionFilters: state.actionFilters.map((group, i) => {
+      if (i !== groupIndex) {
+        return group;
+      }
+      return {
+        ...group,
+        actions: [
+          ...(group.actions ?? []),
+          {
+            type: actionType,
+            data: {},
+          },
+        ],
+      };
+    }),
+  };
+}
+
+function removeIfAction(
+  state: AutomationBuilderState,
+  action: RemoveIfActionAction,
+  formModel?: FormModel
+): AutomationBuilderState {
+  const {groupIndex, actionIndex} = action;
+  if (formModel) {
+    for (const key of formModel.fields.keys()) {
+      if (key.startsWith(`actionFilters.${groupIndex}.actions.${actionIndex}.`)) {
+        formModel.removeField(key);
+      }
+    }
+  }
+  return {
+    ...state,
+    actionFilters: state.actionFilters.map((group, i) => {
+      if (i !== groupIndex) {
+        return group;
+      }
+      return {
+        ...group,
+        actions: group.actions?.filter((_, j) => j !== actionIndex),
+      };
+    }),
+  };
+}
+
+function updateIfAction(
+  state: AutomationBuilderState,
+  action: UpdateIfActionAction
+): AutomationBuilderState {
+  const {groupIndex, actionIndex, data} = action;
+  // special case for integrationId which is outside of data
+  if ('integrationId' in data) {
```

Comment:
thoughts on using react query to format the UI data so this isn't a special case? if you use a pre / post send formatter you could translate the API response and store that in the query cache, then when it goes to send data, it'll automatically re-format it for the API.

Reply:
I'd like to leave this as-is for this PR, but I will keep this idea in mind when adding in all of the other actions since there might be special cases for those as well (still not the most familiar with the formatting needed for the API).